src/skagent/ann.py

```diff
@@ -498,26 +496,56 @@ def value_function(self, states_t, shocks_t={}, parameters={}):
         Compute value function estimates for given state variables.
 
         The value function takes the same information as the policy function
-        (the control's information set) but doesn't need to compute transitions.
+        (the control's information set) and computes derived variables as needed.
 
         Parameters
         ----------
         states_t : dict
             State variables as dict (e.g., {"wealth": tensor(...)})
         shocks_t : dict, optional
-            Shock variables as dict (not used but kept for interface consistency)
+            Shock variables as dict
         parameters : dict, optional
-            Model parameters (not used but kept for interface consistency)
+            Model parameters
 
         Returns
         -------
         torch.Tensor
             Value function estimates
         """
-        # The inputs to the network are the information set variables
-        # Combine states_t and shocks_t to get all available variables
-        all_vars = states_t | shocks_t
-        iset_vals = [all_vars[isym].flatten() for isym in self.cobj.iset]
+        # Follow the same pattern as BlockPolicyNet.decision_function
+        vals = parameters | states_t | shocks_t
+
+        # Try to compute derived variables using block transition
+        # If variables are missing for complex models, fall back to available variables
```

Comment:
I don't understand the context where these fallbacks are used.

Are they needed for the Bellman training loop?

Are there separate automated tests that test for the execution of these code blocks?

Maybe these can be moved to a separate PR?